prov/verbs/src/fi_verbs.c

```diff
@@ -815,25 +792,63 @@ static int fi_ibv_init_info(const struct fi_info *hints)
 	default:
 		FI_INFO(&fi_ibv_prov, FI_LOG_CORE, "Unknown transport type");
 		ret = -FI_ENODATA;
-		goto err3;
+		goto err;
 	}
 
 	if (!(fi->domain_attr->name = strdup(ctx->device->name))) {
 		ret = -FI_ENOMEM;
-		goto err3;
+		goto err;
 	}
 
-	verbs_info = fi;
-	rdma_free_devices(ctx_list);
-unlock:
-	pthread_mutex_unlock(&verbs_info_lock);
+	*info = fi;
 	return 0;
-
-err3:
+err:
 	fi_freeinfo(fi);
-err2:
+	return ret;
+}
+
+static int fi_ibv_init_info(const struct fi_info *hints)
```

Comment:
Ok. In that case, we can just use the default_inline_data variable to set the max inject size. It would be configurable by the user through a file similar to def_send_wr.